layers/+distributions/spacemacs-base/local/centered-buffer-mode/centered-buffer-mode.el

```diff
@@ -0,0 +1,190 @@
+  (-(/ (- (window-pixel-width window)
+          (or text-pixel-size
+              (car (window-text-pixel-size window))))
+       2)
+    (if (bound-and-true-p org-indent-mode) 40
+      spacemacs-centered-buffer-mode-safety-gap-width)))
+
+(defun spacemacs//centered-buffer-buffer-list-update-fringes ()
+  "Used in `spacemacs//centered-buffer-buffer-list-update-hook' and
+`spacemacs//centered-buffer-buffer-window-configuration-change-hook'."
+  (dolist (frame (frame-list))
+    (when (frame-live-p frame)
+      (dolist (window (window-list frame 2))
+        (when (and (buffer-local-value 'spacemacs--centered-buffer-mode-origin-buffer
+                                       (window-buffer window))
+                   ;; Might be needed because
+                   ;; (spacemacs-centered-buffer-mode -1) kills buffers.
+                   (buffer-live-p (window-buffer window)))
+          (let ((fringe-w (spacemacs//centered-buffer-calc-fringe
+                           window
+                           spacemacs--centered-buffer-mode-text-pixel-size)))
+            (if (> fringe-w spacemacs-centered-buffer-mode-min-fringe-width)
+                (set-window-fringes window fringe-w fringe-w t)
+              (spacemacs-centered-buffer-mode -1)))))))
+  ;; Prevent premature evaluation.
+  (when spacemacs--centered-buffer-mode-indirect-buffers
+    (spacemacs//centered-buffer-prune-indirect-buffer-list)))
+
+(defun spacemacs//centered-buffer-prune-indirect-buffer-list ()
+  "Remove indirect buffer from the `spacemacs--centered-buffer-mode-indirect-buffers'
+if it not displayed. Disables `centered-buffer-mode' hooks
+if `spacemacs--centered-buffer-mode-indirect-buffers' has no elements left(nil)."
+  (dolist (buffer spacemacs--centered-buffer-mode-indirect-buffers)
+   (unless (and (buffer-live-p buffer)
+                (window-live-p (get-buffer-window buffer t)))
+      (delete buffer 'spacemacs--centered-buffer-mode-indirect-buffers)
+      (let ((origin-buffer (buffer-local-value
+                            'spacemacs--centered-buffer-mode-origin-buffer
+                            buffer)))
+        (when (buffer-live-p origin-buffer)
+          (with-current-buffer origin-buffer
+            (setq spacemacs--centered-buffer-mode-indirect-buffers nil))))
+      (kill-buffer buffer)))
+  (unless spacemacs--centered-buffer-mode-indirect-buffers
+    (remove-hook 'buffer-list-update-hook
+                 'spacemacs//centered-buffer-buffer-list-update-fringes)
+    (remove-hook 'window-configuration-change-hook
+                 'spacemacs//centered-buffer-buffer-list-update-fringes)))
+
+(defmacro with-mode-disabled (mode &rest body)
+  "Evaluate BODY with MODE disabled."
+  (declare (indent 1) (debug t))
+  `(if (not (bound-and-true-p ,mode))
+       (progn ,@body)
+     (,mode -1)
+     ,@body
+     (,mode +1)))
+
+(provide 'centered-buffer-mode)
+
+;;; centered-buffer-mode.el ends here
```

Comment:
why not extract your minor mode into its own repo and put it on melpa -- in the meantime, you can use quelpa??